blockchain/block_storage.go

```diff
@@ -0,0 +1,61 @@
+package blockchain
+
+import (
+	"encoding/binary"
+
+	"github.com/NethermindEth/juno/core"
+	"github.com/NethermindEth/juno/core/felt"
+	"github.com/NethermindEth/juno/db"
+	"github.com/fxamacker/cbor/v2"
+)
+
+// BlockStorage is just a dumb storage that allows saving and retrieving blocks from database
+type BlockStorage struct {
+	txn db.Transaction
+}
+
+func NewBlockStorage(txn db.Transaction) *BlockStorage {
+	return &BlockStorage{txn: txn}
+}
+
+// Put stores the given block in the database. No check on whether the hash matches or not is done
+func (s *BlockStorage) Put(block *core.Block, hash *felt.Felt) error {
+	var numBytes [8]byte
+	binary.BigEndian.PutUint64(numBytes[:], block.Number)
+	if err := s.txn.Set(db.BlockNumbersByHash.Key(hash.Marshal()), numBytes[:]); err != nil {
+		return err
+	}
+
+	if blockBytes, err := cbor.Marshal(block); err != nil {
+		return err
+	} else if err = s.txn.Set(db.BlocksByNumber.Key(numBytes[:]), blockBytes); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// GetByNumber retrieves a block from database by its number
```

Comment:
The PR looks good to me. 

Just curious about the naming. Is there any specific reason why `GetByNumber` is preferred as opposed to `GetBlockByNumber`?

Reply:
My thought was; since this is `BlockStorage`, the only thing one can get from it is a block.

Reply:
Got it. 

It's probably just the way my brain is wired. But I think it might not be that obvious once it's used outside `BlockStorage`. 

Since the module name is `blockchain`, it may not be that obvious when it's used like `blockchain.GetByNumber(...`. There are some other things mentioned in Blockchain as well. Like transactions, transaction receipts, etc.

Reply:
a `BlockStorage` instance shouldnt be named `blockchain` probably